definition/definition_ser.go

```diff
@@ -401,152 +405,59 @@ func createLink(tasks map[string]*Task, linkRep *LinkRep, id int, ef expression.
-			if ok && len(conditionStr) > 0 && conditionStr[0] == '=' {
-				conditionStr = conditionStr[1:]
-			}
-			conditionExpr, err := ef.NewExpr(conditionStr)
-			if err != nil {
-				return nil, fmt.Errorf("compile doWhile condition error: %s", err.Error())
-			}
-			loop.condition = conditionExpr
-			delete(settings, "condition")
-		}
+}
 
-		iterate, exist := settings["iterate"]
-		if exist && iterate != nil {
-			iterateStr, ok := iterate.(string)
-			if ok && iterateStr[0] == '=' {
-				iterateStr = iterateStr[1:]
-				iterateStrExpr, err := ef.NewExpr(iterateStr)
-				if err != nil {
-					return nil, fmt.Errorf("compile doWhile condition error: %s", err.Error())
-				}
-				loop.iterate = iterateStrExpr
-			} else {
-				loop.iterate = iterate
-			}
-			delete(settings, "iterate")
-		}
+func getLoopCfg(settings map[string]interface{}, ef expression.Factory) (*loop, error) {
+	loopCfg := &loopCfg{}
 
-		//Accumuate output for loop
-		accumulateOutput, ok := settings["accumulate"]
-		if ok {
-			accumulated, _ := coerce.ToBool(accumulateOutput)
-			if accumulated {
-				//Make sure in loop otherwise it doesn't make sense to enable accumulate
-				if !loop.DowhileEnabled() && !loop.IterateEnabled() {
-					return nil, fmt.Errorf("accumulate only for iterate and dowhile task")
-				}
-			}
-			loop.accumulate = accumulated
-			delete(settings, "accumulate")
+	loopCfgMap, err := coerce.ToObject(settings["loopConfig"])
+	if err != nil {
+		return nil, err
+	}
+
+	if loopCfgMap != nil {
+		
+		err = metadata.MapToStruct(loopCfgMap, loopCfg, true)
+
+		if err != nil {
+			return nil, err
 		}
 
-		retryOnError, ok := settings["retryOnError"]
-		if ok && retryOnError != nil {
-			retryObj, err := coerce.ToObject(retryOnError)
-			if err != nil {
-				return nil, fmt.Errorf("retryOnError configuration error : %s", err.Error())
-			}
+	} else {
 
-			count, exist := retryObj["count"]
-			if exist && count != nil {
-				strVal, ok := count.(string)
-				if ok && len(strVal) > 0 && strVal[0] == '=' {
-					count, err = resolve.Resolve(strVal[1:], nil)
-					if err != nil {
-						return nil, err
-					}
-				}
-				cnt, err := coerce.ToInt(count)
-				if err != nil {
-					return nil, fmt.Errorf("retryOnError count must be int")
-				}
-				loop.retryOnError.count = cnt
-			}
+		err = metadata.MapToStruct(settings, loopCfg, true)
 
-			interval, exist := retryObj["interval"]
-			if exist && interval != nil {
-				strVal, ok := interval.(string)
-				if ok && len(strVal) > 0 && strVal[0] == '=' {
-					interval, err = resolve.Resolve(strVal[1:], nil)
-					if err != nil {
-						return nil, err
-					}
-				}
-				intervalInt, err := coerce.ToInt(interval)
-				if err != nil {
-					return nil, fmt.Errorf("retryOnError interval must be int")
-				}
-				loop.retryOnError.interval = intervalInt
-			}
-			delete(settings, "retryOnError")
+		if err != nil {
+			return nil, err
+		}
+		dowhileObj, err := coerce.ToObject(settings["doWhile"])
```

Comment:
@lixingwang this should handle the old "doWhile" case.

Reply:
Got it.  I realized that thanks.